datasets/speechocean762/speechocean762.py

```diff
@@ -0,0 +1,174 @@
+    """
+
+    VERSION = datasets.Version("1.2.0")
+
+    def _info(self):
+        features = datasets.Features(
+            {
+                "id": datasets.Value("string"),
+                "speaker": datasets.Value("string"),
+                "age": datasets.Value("int32"),
+                "gender": datasets.Value("string"),
+                "text": datasets.Value("string"),
+                "audio": datasets.Audio(sampling_rate=_SAMPLE_RATE),
+                "accuracy": datasets.Value("float32"),
+                "completeness": datasets.Value("float32"),
+                "fluency": datasets.Value("float32"),
+                "prosodic": datasets.Value("float32"),
+                "total": datasets.Value("float32"),
+                "words": datasets.Sequence(
+                    {
+                        "text": datasets.Value("string"),
+                        "accuracy": datasets.Value("float32"),
+                        "stress": datasets.Value("float32"),
+                        "total": datasets.Value("float32"),
+                        "phones": datasets.Sequence(datasets.Value("string")),
+                        "phones-accuracy": datasets.Sequence(datasets.Value("float32")),
+                        "mispronunciations": datasets.Sequence(
+                            {
+                                "index": datasets.Value("int32"),
+                                "canonical-phone": datasets.Value("string"),
+                                "pronounced-phone": datasets.Value("string"),
+                            }
+                        ),
+                    }
+                ),
+                "original_full_path": datasets.Value("string"),  # relative path to full audio in original data dirs
+            }
+        )
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        archive_path = dl_manager.download(_DL_URL)
+        local_extracted_archive = dl_manager.extract(archive_path) if not dl_manager.is_streaming else {}
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={"local_extracted_archive": local_extracted_archive, "split": "train"},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={"local_extracted_archive": local_extracted_archive, "split": "test"},
+            ),
+        ]
+
+    def _generate_examples(self, local_extracted_archive, split):
+        root_path = Path(local_extracted_archive) / "speechocean762"
+        meta_path = root_path / f"{split}.json"
+
+        with open(meta_path, encoding="utf-8") as f:
+            info = json.load(f)
+            for utt_id in info:
```

Comment:
and here you need to iterate over the files of the TAR archive
```python
for path, f in files:
    utt_id = path.split("/")[-1].split(".")[0]
    audio_info = info[utt_id]
    audio_info["audio"] = {"path": path, "bytes": f.read()}
    yield utt_id, audio_info
```

note that you may have to host the JSON file somewhere else, to load it separately.

Reply:
An example from Gigaspeech on how this might be achieved: https://huggingface.co/datasets/speechcolab/gigaspeech/blob/main/gigaspeech.py#L345

Reply:
Also make sure you get a copy of audio sample metadata with `audio_info = dict(info[utt_id])` instead of `audio_info = info[utt_id]` in order not to fill the original `info` metadata dict with audio bytes to avoid increasing RAM usage.